caddyhttp/httpserver/middleware.go

```diff
@@ -48,20 +47,48 @@ type (
 
 	// RequestMatcher checks to see if current request should be handled
 	// by underlying handler.
-	//
-	// TODO The long term plan is to get all middleware implement this
-	// interface and have validation done before requests are dispatched
-	// to each middleware.
 	RequestMatcher interface {
 		Match(r *http.Request) bool
 	}
+
+	// Config is a middleware configuration.
+	// This makes it possible for middlewares to have a common
+	// configuration interface.
+	//
+	// TODO The long term plan is to get all middleware implement this
+	// interface for configurations.
+	Config interface {
+		RequestMatcher
+		BasePath() string
```

Comment:
You're right, but what if we don't limit this to the base path in the future? In the end, it's a sorting problem, and the highest-sorted config gets dibs. :) I think we should look for a more general way to handle this.